src/NuGet.Clients/NuGet.PackageManagement.VisualStudio/Projects/VsProjectAdapter.cs

```diff
@@ -230,13 +230,13 @@ private async Task<string> GetTargetFrameworkStringAsync()
             await _threadingService.JoinableTaskFactory.SwitchToMainThreadAsync();
 
             var projectPath = FullName;
-            var platformIdentifier = BuildProperties.GetPropertyValue(
+            var platformIdentifier = await BuildProperties.GetPropertyValueAsync(
                 ProjectBuildProperties.TargetPlatformIdentifier);
-            var platformVersion = BuildProperties.GetPropertyValue(
+            var platformVersion = await BuildProperties.GetPropertyValueAsync(
                 ProjectBuildProperties.TargetPlatformVersion);
-            var platformMinVersion = BuildProperties.GetPropertyValue(
+            var platformMinVersion = await BuildProperties.GetPropertyValueAsync(
                 ProjectBuildProperties.TargetPlatformMinVersion);
-            var targetFrameworkMoniker = BuildProperties.GetPropertyValue(
+            var targetFrameworkMoniker = await BuildProperties.GetPropertyValueAsync(
```

Comment:
This will cause a performance regression that was fixed by https://github.com/NuGet/NuGet.Client/pull/4690. From that PR's description:

> With the current implementation we end up nesting async/JTF.Run unnecessarily.
> This is slower and causes lots of allocations.
> 
> Small changes like this can go a long way.

Unfortunately, async performance is nuanced, so blanket rules like "always use async APIs" will harm performance in some scenarios.

This comment applies to all files in the PR where `GetPropertyValue` was changed to `GetPropertyValueAsync`.